File: src/main/java/RailgunDownloaderV4/Components/YoutubeUI/QualityList.java
package RailgunDownloaderV4.Components.YoutubeUI;

import javax.swing.*;
import java.awt.*;

public class QualityList {
    public void SetQuality(JFrame App, JList<String> QualityListBox) {
        String[] QualityItems = {
                "best",
                "worst",
                "bestvideo",
                "worstvideo",
                "bestaudio",
                "worstaudio",
                "1080p",
                "720p",
                "480p",
                "360p",
                "144p"
        };

        QualityListBox.setListData(QualityItems);
        QualityListBox.setSelectionMode(ListSelectionModel.SINGLE_SELECTION);
        QualityListBox.setSize(new Dimension(70, 180));
        QualityListBox.setLocation(new Point(330, 420));
        QualityListBox.setBackground(new Color(21, 21, 21));
        QualityListBox.setForeground(Color.WHITE);
        QualityListBox.setSelectedIndex(0);
        QualityListBox.setFont(new Font("Consolas", Font.PLAIN, 10));

        QualityListBox.setToolTipText("Choose your video quality");

        QualityListBox.setCellRenderer(new DefaultListCellRenderer() {
            @Override
            public Component getListCellRendererComponent(JList<?> list, Object value, int index, boolean isSelected, boolean cellHasFocus) {
                JLabel QualityListLabel = (JLabel) super.getListCellRendererComponent(list, value, index, isSelected, cellHasFocus);

                QualityListLabel.setHorizontalAlignment(SwingConstants.CENTER);
                return QualityListLabel;
            }
        });

        App.add(QualityListBox);
    }
}

File: src/main/kotlin/railgunDownloaderV4/components/youtubeUI/QualityList.kt
/*
 * RAILGUN DOWNLOADER - VERSION 4.0.0
 * File: QualityList.kt
 * This project is licensed GPL-3.0 License.
 * LICENSE: https://www.gnu.org/licenses/gpl-3.0.html
 * Contribution: Reim
 */
package railgunDownloaderV4.components.youtubeUI

import java.awt.*
import javax.swing.*

class QualityList {
    fun setQuality(app: JFrame, qualityListBox: JList<String>) {
        val qualityItems = arrayOf(
            "best",
            "worst",
            "bestvideo",
            "worstvideo",
            "bestaudio",
            "worstaudio"
        )

        qualityListBox.setListData(qualityItems)
        qualityListBox.selectionMode = ListSelectionModel.SINGLE_SELECTION
        qualityListBox.size = Dimension(70, 180)
        qualityListBox.location = Point(330, 420)
        qualityListBox.background = Color(21, 21, 21)
        qualityListBox.foreground = Color.WHITE
        qualityListBox.selectedIndex = 0
        qualityListBox.font = Font("Consolas", Font.PLAIN, 10)

        qualityListBox.toolTipText = "Choose your video quality"

        qualityListBox.cellRenderer = object : DefaultListCellRenderer() {
            override fun getListCellRendererComponent(
                list: JList<*>?,
                value: Any,
                index: Int,
                isSelected: Boolean,
                cellHasFocus: Boolean
            ): Component {
                val qualityLabel = super.getListCellRendererComponent(list, value, index, isSelected, cellHasFocus) as JLabel

                qualityLabel.horizontalAlignment = CENTER
                return qualityLabel
            }
        }

        app.add(qualityListBox)
    }
}
